src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -49,7 +49,8 @@ GcsActorScheduler::GcsActorScheduler(
 void GcsActorScheduler::Schedule(std::shared_ptr<GcsActor> actor) {
   RAY_CHECK(actor->GetNodeID().IsNil() && actor->GetWorkerID().IsNil());
 
-  if (RayConfig::instance().gcs_actor_scheduling_enabled()) {
+  if (RayConfig::instance().gcs_actor_scheduling_enabled() &&
+      !actor->GetCreationTaskSpecification().GetRequiredResources().IsEmpty()) {
```

Comment:
This condition is a little weird here.

Reply:
This is just a quick way to make the empty-resource actors go through the legacy logic (randomly select a forward raylet).